src/main/java/ch/interlis/ioxwkf/shp/ShapeWriter.java

```diff
@@ -652,20 +661,49 @@ public void setModel(TransferDescription td) {
 	}
 	
 	public AttributeDescriptor[] getAttributeDescriptors() {
-		return attrDescs.toArray(new AttributeDescriptor[attrDescs.size()]);
+        List<AttributeDescriptor> attrDescs = new ArrayList<AttributeDescriptor>();
+        for (Map.Entry<String, AttributeDescriptor> entry : attrDescsMap.entrySet()) {
+            attrDescs.add(entry.getValue());
+        }
+        return attrDescs.toArray(new AttributeDescriptor[attrDescs.size()]);
 	}
 	
 	/** Sets the attribute descriptors of the shape file.
 	 * Alternative to setting a model.
 	 * @param attrDescs[]
 	 */
 	public void setAttributeDescriptors(AttributeDescriptor attrDescs[]) {
-		this.attrDescs = new ArrayList<AttributeDescriptor>();
-		for(AttributeDescriptor attrDesc:attrDescs) {
-			if(attrDesc.getType() instanceof GeometryType) {
-				iliGeomAttrName=attrDesc.getLocalName();
-			}
-			this.attrDescs.add(attrDesc);
-		}
+        this.attrDescsMap = new HashMap<String,AttributeDescriptor>(); 
+        for(AttributeDescriptor attrDesc:attrDescs) {
+            if(attrDesc.getType() instanceof GeometryType) {
+                iliGeomAttrName=attrDesc.getLocalName();
+            }
+            this.attrDescsMap.put(attrDesc.getLocalName(), attrDesc);
+        }
 	}
+	
+    private String trimAttributeName(String attrName) {
+        // Geometry attribute names do not need to be trimmed,
+        // since they are treated specifically.
+        if (attrName.length() <= 10 || attrName.equalsIgnoreCase(iliGeomAttrName)) {
+            return attrName;
+        }
+        // It works for a max of 9 similar/identical names.
+        // Can be extended but at the end of the day it is 
+        // not working endlessly.
+        for (int i=0; i<=9; i++) {
```

Comment:
Nicht gut. Oft haben Namen einen signifikatne Prefix oder Suffix. Besser ist ili2db:NameMapping.shortcutName(). Müsste ich verschieben in ehibasics oder iox-ili, damit es hier verfügbar wird.

Reply:
Hab es auf NameUtility.shortcutName() umgestellt. Das garantiert aber keine Sicherheit vor Kollisionen: Die Attributnamen `SehrLangerText`, `SehrLangerLangerText ` und `SehrLangerLangerLangerText` werden identisch gekürzt. Ist aber eventuell auch ein unglückliches Beispiel, das in Realität kaum vorkommt.